zipline/finance/performance/period.py

```diff
@@ -72,41 +72,41 @@
 
 from __future__ import division
 import logbook
+from operator import mul
 
 import numpy as np
 import pandas as pd
 from collections import (
     defaultdict,
-    OrderedDict,
 )
+
+try:
+    # optional cython based OrderedDict
+    from cyordereddict import OrderedDict
+except ImportError:
+    from collections import OrderedDict
+
 from six import iteritems, itervalues
 
 import zipline.protocol as zp
-from . position import positiondict
 
 log = logbook.Logger('Performance')
+TRADE_TYPE = zp.DATASOURCE_TYPE.TRADE
 
+def position_proxy(func):
+    def _proxied(self, *args, **kwargs):
+        meth_name = func.__name__
+        meth = getattr(self.position_tracker, meth_name)
+        return meth(*args, **kwargs)
+    return _proxied
 
-class FastSeries(object):
-    def __init__(self, *args, **kwargs):
-        super(FastSeries, self).__init__(*args, **kwargs)
 
-        self._loc_map = {}
-        self.series = pd.Series([])
-        self.values = self.series.values
+def proxy_error():
+    import inspect
 
-    def __setitem__(self, key, value):
-        try:
-            i = self._loc_map[key]
-            self.values[i] = value
-        except (KeyError, IndexError):
-            self.series = \
-                self.series.append(
-                    pd.Series({key: value}))
-            self._loc_map = dict(
-                zip(self.series.index,
-                    range(len(self.series))))
-            self.values = self.series.values
+    meth_name = inspect.stack()[1][3]
+    msg = "{meth_name} should have been proxied to position_tracker."
+    return Exception(msg.format(meth_name=meth_name))
```

Comment:
Could we make this a specific exception, maybe `ProxyError` that does the inspect stuff?